src/tests/tests/background.js

```diff
@@ -119,4 +124,22 @@
     }
   });
 
+  QUnit.test("DNT checking obeys user setting", (assert) => {
+    const NUM_TESTS = 5;
+
+    let done = assert.async(NUM_TESTS);
+
+    assert.expect(NUM_TESTS);
+
+    badger.isCheckingDNTPolicyEnabled = () => false;
```

Comment:
OK now I think we are running into an annoying async test issue. I'm seeing other DNT-related tests break I think because we don't restore `isCheckingDNTPolicyEnabled` at exactly the right time: restoring at the end of the test isn't guaranteed to work because the test is async.